base/src/main/java/proguard/io/DataEntry.java

```diff
@@ -45,6 +45,11 @@ public interface DataEntry
      */
     public long getSize();
 
+    /**
+     * Returns the modification time of this data entry in milliseconds,
+     * since the epoch (1970-01-01T00:00:00Z), or -1 if unknown.
+     */
+    public long getModificationTime();
```

Comment:
indeed, I did not think about the case where consumers of the library would implement their own DataEntry classes. Returning -1 by default makes sense ofc.

Could you take a look at the ZipWriter? There I changed the default for the no-additional-arg constructor for the modificationTime parameter from 0 to -1, which means that by default the ZipWriter will pass entries through instead of setting their modification time to 0 which imho is a more reasonable default for consumers of this library.

In case you wanna hide the modification time, e.g. for use-cases like obfuscation, you need to explicitly set it to 0.
I believe, even in the case of ProGuard, it would make sense that by default you would pass the modification time through and only in case of obfuscation enabled, you change it to some hard-coded value like 0.